test/jdk/java/lang/StableValue/StableFieldUpdaterExampleTest.java

```diff
@@ -0,0 +1,190 @@
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/* @test
+ * @summary Basic tests for StableFieldUpdaterExampleTest examples in javadoc
+ * @modules java.base/jdk.internal.lang.stable
+ * @run junit StableFieldUpdaterExampleTest
+ */
+
+import jdk.internal.lang.stable.StableFieldUpdater;
+import org.junit.jupiter.api.Test;
+
+import java.lang.Override;
+import java.util.Objects;
+import java.util.function.ToIntFunction;
+import java.util.function.ToLongFunction;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+final class StableFieldUpdaterExampleTest {
+
+    @interface Stable {} // No access to the real @Stable
+
+    public static final class Bar extends Base {
+        public Bar(int hash) {
+            super(hash);
+        }
+    }
+
+    public static final class Baz extends Base {
+        public Baz(int hash) {
+            super(hash);
+        }
+    }
+
+    public static abstract class Base {
+
+        final int hash;
+
+        public Base(int hash) {
+            this.hash = hash;
+        }
+
+        @Override
+        public boolean equals(Object obj) {
+            return obj instanceof Base that &&
+                    this.getClass().equals(that.getClass()) &&
+                    this.hash == that.hash;
+        }
+
+        @Override
+        public int hashCode() {
+            return hash;
+        }
+
+        @Override
+        public String toString() {
+            return getClass().getSimpleName() + "[hash=" + hash + ']';
+        }
+    }
+
+
+    static // Intentionally in unblessed order to allow the example to look nice
+
+    public final class Foo {
+
+        private final Bar bar;
+        private final Baz baz;
+
+        public Foo(Bar bar, Baz baz) {
+            this.bar = bar;
+            this.baz = baz;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            return o instanceof Foo that &&
+                    Objects.equals(this.bar, that.bar) &&
+                    Objects.equals(this.baz, that.baz);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(bar, baz);
+        }
+    }
+
+    static
+
+    public final class LazyFoo {
+
+        private final Bar bar;
+        private final Baz baz;
+
+        private static final ToIntFunction<LazyFoo> HASH_UPDATER =
+                StableFieldUpdater.ofInt(LazyFoo.class, "hash",
+                        l -> Objects.hash(l.bar, l.baz), -1);
+
+        @Stable
+        private int hash;
+
+        public LazyFoo(Bar bar, Baz baz) {
+            this.bar = bar;
+            this.baz = baz;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            return o instanceof Foo that &&
+                    Objects.equals(this.bar, that.bar) &&
+                    Objects.equals(this.baz, that.baz);
+        }
```

Comment:
```suggestion
        public boolean equals(Object o) {
            return o instanceof Foo that
                    && Objects.equals(this.bar, that.bar)
                    && Objects.equals(this.baz, that.baz);
        }
```
In most cases in JDK code, the operator `&&` is placed in front.